include/arch/riscv/arch/32/mode/hardware.h

```diff
@@ -75,7 +75,7 @@ static inline uint64_t riscv_read_time(void)
         "rdtime  %1\n"
         "rdtimeh %2\n"
         : "=r"(nH1), "=r"(nL), "=r"(nH2));
-    if (nH1 < nH2) {
+    if (nH1 != nH2) {
```

Comment:
A 64-bit rollover of the kernel timer is not allowed to happen. It would be better to add a condition that halts if the nH1 > nH2 condition is seen.

Reply:
I wonder if this is something that should not be checked here, but at the higher layers in a generic way. The timer design for each platforms should be, that this does never roll over practically. This function just returns whatever is in the registers.